### PR TITLE
Add function to ensure that consensus will never be IgnoreConsensus in production

### LINKED ISSUE #792: Ensure `ConsensusUpgrade::IgnoreConsensus` is not used in production code
It should be impossible to use `ConsensusUpgrade::IgnoreConsensus` anywhere other than in unit tests.

### PR DESCRIPTION
**Pull request by TheQuantumPhysicist**
Closes #792 

Please check if I forgot any corner case.

### CODE REVIEW

**Review by azarovh — APPROVED** [review]

File: common/src/chain/config/mod.rs [hunk 1]
```diff
@@ -637,6 +637,40 @@ pub fn create_unit_test_config() -> ChainConfig {
...
+    let all_upgrades = upgrades.all_upgrades();
+
+    for upgrade in all_upgrades {
+        let upgrade_height = &upgrade.0;
+        let upgrade_data = &upgrade.1;
+
+        // We skip genesis
+        if *upgrade_height == 0.into() && all_upgrades.len() > 1 {
+            continue;
+        }
```
Comment (iljakuklic) [inline common/src/chain/config/mod.rs:660]: Is the first upgrade required to be at genesis? If so, we can just assert the first upgrade is at height 0 and only iterate over `all_upgrades.into_iter().skip(1)`.

File: common/src/chain/upgrades/netupgrade.rs [hunk 2]
```diff
@@ -42,6 +42,19 @@ impl NetUpgrades<UpgradeVersion> {
         )])
     }
 
+    pub fn deliberate_ignore_consensus_twice() -> Self {
+        Self(vec![
```
Comment (iljakuklic) [inline common/src/chain/upgrades/netupgrade.rs:46]: As far as I can see, this is only used in tests. Should it be guarded by `#[cfg(test)]`?

**Review by iljakuklic — APPROVED** [review]

File: common/src/chain/upgrades/netupgrade.rs (hunk 2 above)
Comment (TheQuantumPhysicist) [reply common/src/chain/upgrades/netupgrade.rs:46]: Done

File: common/src/chain/upgrades/netupgrade.rs [hunk 3]
```diff
@@ -57,6 +70,10 @@ impl NetUpgrades<UpgradeVersion> {
             ),
         ])
     }
+
+    pub fn all_upgrades(&self) -> &Vec<(BlockHeight, UpgradeVersion)> {
+        &self.0
```
[the remark this answers is not in the document]
Comment (TheQuantumPhysicist) [reply common/src/chain/upgrades/netupgrade.rs:75]: Done

File: common/src/chain/config/mod.rs (hunk 1 above)
Comment (TheQuantumPhysicist) [reply common/src/chain/config/mod.rs:660]: Done... despite it being more verbose.

**Review by iljakuklic — APPROVED** [review]

**Review by muursh — APPROVED** [review]
